src/action.ts

```diff
@@ -0,0 +1,60 @@
+import * as github from "@actions/github";
+import { Octokit } from "@octokit/rest";
+import { Value } from "@sinclair/typebox/value";
+import { plugin } from "./plugin";
+import { envSchema, envValidator, PluginInputs, pluginSettingsSchema, pluginSettingsValidator } from "./types";
+
+/**
+ * How a GitHub action executes the plugin.
+ */
+export async function run() {
+  const payload = github.context.payload.inputs;
+
+  payload.env = { ...(payload.env || {}), workflowName: github.context.workflow };
+  if (!envValidator.test(payload.env)) {
+    const errors: string[] = [];
+    for (const error of envValidator.errors(payload.env)) {
+      console.error(error);
+      errors.push(`${error.path}: ${error.message}`);
+    }
+    throw new Error(`Invalid environment provided:\n${errors.join(";\n")}`);
+  }
+  const env = Value.Decode(envSchema, payload.env || {});
+
+  payload.settings = Value.Default(pluginSettingsSchema, JSON.parse(payload.settings));
+  if (!pluginSettingsValidator.test(payload.settings)) {
+    const errors: string[] = [];
+    for (const error of pluginSettingsValidator.errors(payload.settings)) {
+      console.error(error);
+      errors.push(`${error.path}: ${error.message}`);
+    }
+    throw new Error(`Invalid settings provided:\n${errors.join(";\n")}`);
+  }
+
+  const settings = Value.Decode(pluginSettingsSchema, payload.settings);
+  const inputs: PluginInputs = {
+    stateId: payload.stateId,
+    eventName: payload.eventName,
+    eventPayload: JSON.parse(payload.eventPayload),
+    settings,
+    authToken: payload.authToken,
+    ref: payload.ref,
+  };
+
+  await plugin(inputs, env);
+
+  return returnDataToKernel(inputs.authToken, inputs.stateId, {});
```

Comment:
Here you are passing auth token for the repo that triggered this event to the function that returns data to the kernel however this token should be for plugin's repo (`automated-merging`)
I think this is the source of https://github.com/ubiquibot/automated-merging/pull/1#issuecomment-2231237542 problem

Reply:
This is the token that is provided by the Kernel to post on behalf on Ubiquibot, so is that not correct, and it should be the github token that is provided by the action inside the env (GITHUB_TOKEN), am I understanding right?

Reply:
yes it should be github token provided by the action because you are making repository dispatch event on `automated-merging` so the token provided by the kernel doesn't have permission for this repo (only for the repo that triggered the kernel)